### PR TITLE
12.0 remove website dependency for recaptcha

### PR DESCRIPTION
**Pull request by victor-champonnois**
related to [this issue](https://github.com/coopiteasy/vertical-cooperative/issues/214)
[related pr](https://github.com/coopiteasy/addons/pull/218)

### CODE REVIEW

**Comment by victor-champonnois** [comment]
@robinkeunen @remytms Ready to review !

File: website_recaptcha/models/portal_mixin.py [hunk 1]
```diff
@@ -3,17 +3,17 @@
...
 
-class Website(models.Model):
-    _inherit = "website"
+class PortalMixin(models.AbstractModel):
+    _inherit = "portal.mixin"
 
-    recaptcha_key_site = fields.Char()
-    recaptcha_key_secret = fields.Char()
+    # recaptcha_key_site = fields.Char()
+    # recaptcha_key_secret = fields.Char()
```
Comment (robinkeunen) [inline website_recaptcha/models/portal_mixin.py]: delete comments.

File: website_recaptcha/models/res_config.py [hunk 2]
```diff
@@ -4,23 +4,20 @@
...
         "Google reCaptcha",
         compute="_compute_has_google_recaptcha",
         inverse="_inverse_has_google_recaptcha",
         readonly=False,
     )
 
-    @api.depends("website_id")
+    @api.depends("recaptcha_key_site")
     def _compute_has_google_recaptcha(self):
         self.has_google_recaptcha = bool(self.recaptcha_key_site)
```
Comment (robinkeunen) [inline website_recaptcha/models/res_config.py]: Is this really necessary ? Are the fields `recaptcha_key_site` and `recaptcha_key_secret` not enough ?

File: website_recaptcha/views/res_config.xml [hunk 3]
```diff
@@ -2,76 +2,86 @@
...
-                <div id="google_maps_setting" position="after">
+                <xpath expr="//div[@name='integration']" position="after">
+                    <h2>Portal reCaptcha2</h2>
+                    <div class="row o_settings_container mt16">
+                    <!-- <div
+                        class="app_settings_block"
+                        data-string=""
+                        string="Shifts Management"
+                        data-key="beesdoo_shift"
+                        groups="group_cooperative_admin"
```
Comment (robinkeunen) [inline website_recaptcha/views/res_config.xml]: remove comments

File: website_recaptcha/views/res_config.xml [hunk 4]
```diff
@@ -2,76 +2,86 @@
...
-                                class="mr-2"
-                                src="website_form_recaptcha/static/description/recaptcha_ico.png"
-                            />
-                            <label for="has_google_recaptcha" string="reCaptcha" />
+                                    width="20px"
+                                    alt="captcha"
+                                    class="mr-2"
+                                    src="website_recaptcha/static/description/icon.png"
+                                />
+                            <!-- <label for="has_google_recaptcha" string="reCaptcha" />
```
Comment (robinkeunen) [inline website_recaptcha/views/res_config.xml]: same

**Review by robinkeunen — CHANGES_REQUESTED** [review]

File: website_recaptcha/models/res_config.py (hunk 2 above)
Comment (victor-champonnois) [reply website_recaptcha/models/res_config.py]: Yes I wasn't sure whether it is worth keeping that. I removed it.

**Pull request closed without merging**